### PR TITLE
Clarify requirements in README.md

### PR DESCRIPTION
**Pull request by apelisse**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,26 @@
...
+# Style guide
+
+Here is a list of guidelines to follow when submitting code to that repository:
+
+- Full unit-test coverage.
+
+- Go tools compliant (`go get`, `go test`, etc.). It needs to be vendorable
+  somewhere else.
+
+- No dependence on `k8s.io/kubernetes`. This repository can depend on other
```
Comment (monopole) [inline README.md]: No dependence on `k8s.io/kubernetes`.  Dependence on other repositories is fine.

File: README.md [hunk 2]
```diff
@@ -1,6 +1,26 @@
...
+  sub-repositories.
+
+- Libraries must be documented properly.
+
+- Ideally, packages and libraries in this repository have a limited set of
+  dependency, but rather define sensible interfaces.
```
Comment (monopole) [inline README.md]: Packages in this repository should aspire to implement sensible, small interfaces and import a limited set of dependencies.

File: README.md [hunk 3]
```diff
@@ -1,6 +1,26 @@
...
+
+- Full unit-test coverage.
+
+- Go tools compliant (`go get`, `go test`, etc.). It needs to be vendorable
+  somewhere else.
+
+- No dependence on `k8s.io/kubernetes`. This repository can depend on other
+  sub-repositories.
+
+- Libraries must be documented properly.
```
Comment (monopole) [inline README.md]:
```
Code must be usefully [commented](https://golang.org/doc/effective_go.html#commentary).
```
The repository's [pre-commit](https://github.com/kubernetes/kubectl/blob/master/bin/pre-commit.sh) script enforces golint and go vet which check certain comment related things.

File: README.md [hunk 4]
```diff
@@ -1,6 +1,26 @@
...
+
```
Comment (monopole) [inline README.md]: When reviewing PR's, you're encouraged to utilize Go lang's [code review comments](https://github.com/golang/go/wiki/CodeReviewComments) page.

**Comment by apelisse** [comment]
Updated, PTAL

**Pull request merged**